kinto/core/views/batch.py

```diff
@@ -146,6 +146,11 @@ def post_batch(request):
         except httpexceptions.HTTPException as e:
             if e.content_type == 'application/json':
                 resp = e
+                # Since some request in the batch failed, we need to "doom" the parent process
+                # through Pyramid's transaction manager.
+                transaction_manager = request.tm
+                transaction_manager.begin()
+                transaction_manager.doom()
```

Comment:
I think it needs to check the `resp.status` (eg. if 5XX) like does the commit veto (see https://github.com/Pylons/pyramid_tm/blob/ffed3d3e352b800064dbe9e1f0270f11fdda63ff/pyramid_tm/__init__.py#L50) 

Also, why should it only apply to exception with content-type JSON ? Why should we `.begin()` again?

If you want to add some new tests, you'll find some in `core/test_views_transaction.py`

Reply:
Yes, the resp.status check is a good idea. I'm trying to figure out how that should look and there seems to have been some discussion in the past on how a response should respond when a sub-response breaks.

It seems from the following discussions that it should be assumed when a sub-response breaks with a 4XX, the entire batch should come back with a 4XX:

https://github.com/Kinto/kinto/issues/1205
https://github.com/Kinto/kinto/issues/624

I guess I'm confused as the following test seems to contridact that:

https://github.com/Kinto/kinto/blob/382adcdc2dccff7151798bc7d250769dc818d83e/tests/core/test_views_transaction.py#L59

If the former is true, I will need to check that the `resp.status` is 4XX (as this contraint violation is a 409) and 5XX, but not 401 as the test asserts 401s should not effect other responses being committed.

Reply:
As far as the `begin()`, short answer is I used it because it didn't work without it :) As far as my understanding, I agree, it shouldn't be necessary to use it. `abort()` does work without a `begin()`, but I'll look into this and get an answer.